app/javascript/components/UserAssignmentForm.js

```diff
@@ -0,0 +1,54 @@
+import React from "react";
+import PropTypes from "prop-types";
+
+import UserAssignmentFormField from "./UserAssignmentFormField";
+
+class UserAssignmentForm extends React.Component {
+  constructor(props) {
+    super();
+    this.state = props;
+    this.handleAddClick = this.handleAddClick.bind(this);
+    this.handleDeleteClick = this.handleDeleteClick.bind(this);
+  }
+
+  randomUUID() {
+    return "new-mission-"+ Math.floor(Math.random() * Math.floor(9000));
+  }
+
+  handleAddClick() {
+    let assignments = this.state.assignments;
+    assignments.push({role: "", mission: this.randomUUID(), new_assignment: true});
+    this.setState({assignments: assignments});
```

Comment:
See https://stackoverflow.com/questions/23966438/what-is-the-preferred-way-to-mutate-a-react-state for notes on how to properly add to an array in state. You're not supposed to mutate the array directly I think.